### PR TITLE
v23.1.24 release notes (cloud-first)

### PR DESCRIPTION
**Pull request by taroface**
[REL-1024](https://cockroachlabs.atlassian.net/browse/REL-1024)

@florence-crl, this is scheduled for publication tomorrow, July 18.

### CODE REVIEW

File: src/current/_includes/releases/v23.1/v23.1.24.md
Comment (florence-crl) [inline src/current/_includes/releases/v23.1/v23.1.24.md:20]:
```suggestion
- Fixed a bug where a change to a [user-defined type (UDT)]({% link v23.1/create-type.md %}) could cause queries against tables using that type to fail with an error message like: `histogram.go:694: span must be fully contained in the bucket`. The change to the user-defined type could occur either directly from an [`ALTER TYPE`]({% link v23.1/alter-type.md %} statement or indirectly from an [`ALTER DATABASE ... ADD REGION`]({% link v23.1/alter-database.md %}#add-region) or [`ALTER DATABASE ... DROP REGION`]({% link v23.1/alter-database.md %}#drop-region) statement, which implicitly modifies the `crdb_internal_region` UDT. This bug had existed since UDTs were introduced in v20.2. [#124855][#124855]
```

File: src/current/_includes/releases/v23.1/v23.1.24.md
Comment (florence-crl) [inline src/current/_includes/releases/v23.1/v23.1.24.md:23]: maybe add a link on `changefeed.aggregator.flush_jitter` to  https://www.cockroachlabs.com/docs/v23.1/cluster-settings#setting-changefeed-aggregator-flush-jitter
maybe add a link on `min_checkpoint_frequency` to https://www.cockroachlabs.com/docs/v24.1/create-changefeed#min-checkpoint-frequency

File: src/current/_includes/releases/v23.1/v23.1.24.md
Comment (florence-crl) [inline src/current/_includes/releases/v23.1/v23.1.24.md:25]: maybe add a link on "routine" to https://www.cockroachlabs.com/docs/v23.1/user-defined-functions

File: src/current/_includes/releases/v23.1/v23.1.24.md
Comment (florence-crl) [inline src/current/_includes/releases/v23.1/v23.1.24.md:26]: nit
```suggestion
- Fixed the statistics estimation code in the [optimizer]({% link v23.1/cost-based-optimizer.md %}) so it does not use the empty histograms produced if histogram collection has been disabled during statistics collection due to excessive memory utilization. Now the optimizer will rely on distinct counts instead of the empty histograms and should produce better plans as a result. This bug had existed since v22.1. [#126158][#126158]
```

**Review by florence-crl — APPROVED** [review]
LGTM pending minor corrections. I leave it to you whether you add the suggested links.

**Comment by taroface** [comment]
TFTR! I added most of the links.

File: src/current/_data/releases.yml
Comment (florence-crl) [reply src/current/_data/releases.yml:6371]: I saw that you marked this convo as resolved. From my understanding the `lts: true` line should be added for `v23.1.24`. Will you be adding it?

Comment (taroface) [reply src/current/_data/releases.yml:6372]: I forgot to push my changes after I commented here. Yes, I did add the field! Sorry I didn't acknowledge that.

**Pull request merged**
